modules/navier_stokes/test/tests/finite_volume/ins/channel-flow/2d-mixing-length.i

```diff
@@ -0,0 +1,199 @@
+  [mixing_len]
+    order = CONSTANT
+    family = MONOMIAL
+    fv = true
+  []
+[]
+
+[FVKernels]
+  [mass]
+    type = INSFVMassAdvection
+    variable = pressure
+    advected_interp_method = ${advected_interp_method}
+    velocity_interp_method = ${velocity_interp_method}
+    vel = 'velocity'
+    pressure = pressure
+    u = u
+    v = v
+    mu = ${mu}
+    rho = ${rho}
+  []
+
+  [u_advection]
+    type = INSFVMomentumAdvection
+    variable = u
+    advected_quantity = 'rhou'
+    vel = 'velocity'
+    advected_interp_method = ${advected_interp_method}
+    velocity_interp_method = ${velocity_interp_method}
+    pressure = pressure
+    u = u
+    v = v
+    mu = ${mu}
+    rho = ${rho}
+  []
+  [u_viscosity]
+    type = FVDiffusion
+    variable = u
+    coeff = ${mu}
+  []
+  [u_viscosity_rans]
+    type = INSFVMixingLengthReynoldsStress
+    variable = u
+    rho = ${rho}
+    mixing_length = mixing_len
+    momentum_component = 'x'
+    u = u
+    v = v
+  []
+  [u_pressure]
+    type = INSFVMomentumPressure
+    variable = u
+    momentum_component = 'x'
+    p = pressure
+  []
+
+  [v_advection]
+    type = INSFVMomentumAdvection
+    variable = v
+    advected_quantity = 'rhov'
+    vel = 'velocity'
+    advected_interp_method = ${advected_interp_method}
+    velocity_interp_method = ${velocity_interp_method}
+    pressure = pressure
+    u = u
+    v = v
+    mu = ${mu}
+    rho = ${rho}
+  []
+  [v_viscosity]
+    type = FVDiffusion
+    variable = v
+    coeff = ${mu}
+  []
+  [v_viscosity_rans]
+    type = INSFVMixingLengthReynoldsStress
+    variable = v
+    rho = ${rho}
+    mixing_length = mixing_len
+    momentum_component = 'y'
+    u = u
+    v = v
+  []
+  [v_pressure]
+    type = INSFVMomentumPressure
+    variable = v
+    momentum_component = 'y'
+    p = pressure
+  []
+[]
+
+[AuxKernels]
+  [mixing_len]
+    type = WallDistanceMixingLengthAux
+    walls = 'top bottom'
+    variable = mixing_len
+    execute_on = 'initial'
+    von_karman_const = ${von_karman_const}
+  []
+[]
+
+[FVBCs]
+  [inlet-u]
+    type = INSFVInletVelocityBC
+    boundary = 'left'
+    variable = u
+    function = '1'
+  []
+  [inlet-v]
+    type = INSFVInletVelocityBC
+    boundary = 'left'
+    variable = v
+    function = '0'
+  []
+  [walls-u]
+    type = INSFVNoSlipWallBC
+    boundary = 'top bottom'
+    variable = u
+    function = 0
+  []
+  [walls-v]
+    type = INSFVNoSlipWallBC
+    boundary = 'top bottom'
+    variable = v
+    function = 0
+  []
+  [outlet_p]
+    type = INSFVOutletPressureBC
+    boundary = 'right'
+    variable = pressure
+    function = '0'
+  []
+[]
+
+[Materials]
+  [ins_fv]
+    type = INSFVMaterial
+    u = 'u'
+    v = 'v'
+    pressure = 'pressure'
+    rho = ${rho}
+  []
+[]
+
+[Executioner]
+  type = Steady
+  solve_type = 'NEWTON'
+  petsc_options_iname = '-pc_type -ksp_gmres_restart -sub_pc_type -sub_pc_factor_shift_type'
+  petsc_options_value = 'asm      200                lu           NONZERO'
```

Comment:
you might run into the same issue I ran into with those, see #16911

Reply:
What was the issue? Were the results not reproducible on CIVET with ASM?

Reply:
yup, was getting failures in next. Had to use lu

Reply:
Added the `Navier Stokes` testing target here

Reply:
:( yup, got a fail there. Do yall know what the root cause is yet?

Reply:
Under investigation in #16911

Reply:
@smharper your failure is a hard failure unrelated to the work in #16911